heartbeat/monitors/wrappers/monitorstate/monitorstate.go

```diff
@@ -72,10 +75,11 @@ type State struct {
 	Ends            *State `json:"ends"`
 	flappingEnabled bool
 	ctr             int
+	initialDuration int64
 }
 
 func (s *State) incrementCounters(status StateStatus) {
-	s.DurationMs = time.Since(s.StartedAt).Milliseconds()
+	s.DurationMs = time.Since(s.StartedAt).Milliseconds() + s.initialDuration
```

Comment:
@shahzad31 are we taking into account that the agent running this might not be the same as the one which created the state (and initialized `s.initialDuration`? I don't think duration will be consistent in that case, eg: running on `run_once` mode